torch/distributed/rpc.py

```diff
@@ -215,28 +315,17 @@ def rpc(to, func, args=None, kwargs=None, async_call=False):
         >>> result = fut1.wait() + fut2.wait()
         >>> dist.join_rpc()
 
-        One worker 1:
+        On worker 1:
         >>> import torch.distributed as dist
         >>> dist.init_process_group(backend='gloo', rank=1, world_size=2)
         >>> dist.init_model_parallel("worker1")
         >>> dist.join_rpc()
     """
-    if not callable(func):
-        raise TypeError("function should be callable.")
-
-    qualified_name = torch.jit._find_builtin(func)
-
-    args = args if args else ()
-    kwargs = kwargs if kwargs else {}
-
-    if qualified_name is not None:
-        fut = invoke_rpc_builtin(
-            _agent, _to_worker_id(to), qualified_name, *args, **kwargs)
-    else:
-        fut = invoke_rpc_python_udf(
-            _agent, _to_worker_id(to), serialize(PythonUDF(func, args, kwargs)))
-
+    warnings.warn(
+        """dist.rpc is deprecated. Use dist.rpc_async for asynchronous
```

Comment:
nit: shall we modify `test_rpc.py` and `test_dist_autograd` in this PR as well? Otherwise, this warning message will show up in tests?

Reply:
Yep, sounds good